pkg/controllers/sync/controller.go

```diff
@@ -496,16 +494,16 @@ func (s *SyncController) syncToClusters(ctx context.Context, fedResource Federat
 			}
 			if clusterObj.GetDeletionTimestamp() != nil {
 				// Resource is marked for deletion
-				dispatcher.RecordStatus(clusterName, fedtypesv1a1.WaitingForRemoval)
+				dispatcher.RecordStatus(clusterName, fedcorev1a1.WaitingForRemoval)
 				continue
 			}
-			if cluster.GetDeletionTimestamp() != nil && !util.IsCascadingDeleteEnabled(cluster) {
+			if cluster.GetDeletionTimestamp() != nil && !cascadingdeletion.IsCascadingDeleteEnabled(cluster) {
 				// If cluster is terminating and cascading-delete is disabled,
 				// disallow deletion to preserve cluster object.
 				// This could happen right after a cluster is deleted:
```

Comment:
```suggestion
				// Deletion could still happen right after a cluster is deleted:
```
Is this comment wrong? Or am I misunderstanding?

Reply:
This block prevents the said deletion from happening. `This could happen` means `This check will succeed under the following scenario`.

Reply:
Got it